packages/superset-ui-core/src/query/types/QueryFormData.ts

```diff
@@ -124,8 +124,8 @@ export interface BaseFormData extends TimeRange, FormDataResidual {
   groupby?: QueryFormColumn[];
   all_columns?: QueryFormColumn[];
   /** list of filters */
-  adhoc_filters?: AdhocFilter[];
-  extra_filters?: QueryFormExtraFilter[];
+  adhoc_filters?: AdhocFilter[] | null;
```

Comment:
`adhoc_filters` may be null when switching datasets, causing this JS error if you hit Cmd + Enter after switching to a new dataset:

<img width="1078" alt="js-error" src="https://user-images.githubusercontent.com/335541/108271545-d2d78380-7125-11eb-80d4-c7c064c9fb32.png">